### PR TITLE
Update color and cobra libraries

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
Again, read commit by commit

File: go.sum [hunk 1]
```diff
@@ -1,12 +1,26 @@
 cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
+cloud.google.com/go v0.38.0/go.mod h1:990N+gfupTy94rShfmMCWGDn0LpTmnzTp2qbd1dvSRU=
+cloud.google.com/go v0.44.1/go.mod h1:iSa0KzasP4Uvy3f1mN/7PiObzGgflwredwwASm/v6AU=
+cloud.google.com/go v0.44.2/go.mod h1:60680Gw3Yr4ikxnPRS/oxxkBccT6SA1yMk63TGekxKY=
+cloud.google.com/go v0.45.1/go.mod h1:RpBamKRgapWJb87xiFSdk4g1CME7QZg3uwTez+TSTjc=
+cloud.google.com/go v0.46.3/go.mod h1:a6bKKbmY7er1mI7TEI4lsAkts/mkhTSZK8w33B4RAg0=
+cloud.google.com/go/bigquery v1.0.1/go.mod h1:i/xbL2UlR5RvWAURpBYZTtm/cXjCha9lbfbpx4poX+o=
```
Comment (na--) [inline go.sum:8]: I wonder why are these needed? :confused: I checked out your branch and ran this: 
```
$ go mod why cloud.google.com/go/bigquery
# cloud.google.com/go/bigquery
(main module does not need package cloud.google.com/go/bigquery)
$ go mod why cloud.google.com/go         
# cloud.google.com/go
(main module does not need package cloud.google.com/go)
```

:confused:

File: go.sum [hunk 2]
```diff
@@ -16,21 +30,39 @@ github.com/Shopify/toxiproxy v2.1.4+incompatible h1:TKdv8HiTLgE5wdJuEML90aBgNWso
...
+github.com/beorn7/perks v1.0.0/go.mod h1:KWe93zE9D1o94FZ5RNwFwVgaQK1VOXiVxmqh+CedLV8=
+github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kBD4zp0CCIs=
+github.com/bketelsen/crypt v0.0.3-0.20200106085610-5cbc8cc4026c/go.mod h1:MKsuJmJgSg28kpZDP6UIiPt0e0Oz0kqKNGyRaWEPv84=
 github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
+github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/client9/misspell v0.3.4/go.mod h1:qj6jICC3Q7zFZvVWo7KLAzC3yx5G7kyvSDkc90ppPyw=
 github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f/go.mod h1:M8M6+tZqaGXZJjfX53e64911xZQV5JYwmTeXPW+k8Sc=
 github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
+github.com/coreos/bbolt v1.3.2/go.mod h1:iRUV2dpdMOn7Bo10OQBFzIJO9kkE559Wcmn+qkEiiKk=
+github.com/coreos/etcd v3.3.13+incompatible/go.mod h1:uF7uidLiAD3TWHmW31ZFd/JWoc32PjwdhPthX9715RE=
```
Comment (na--) [inline go.sum:54]: Same, why are those in `go.sum` :confused: 
```
$ go mod why github.com/coreos/bbolt
# github.com/coreos/bbolt
(main module does not need package github.com/coreos/bbolt)
$ go mod why github.com/coreos/etcd 
# github.com/coreos/etcd
(main module does not need package github.com/coreos/etcd)
$ rg  'etcd|bbolt' .               
./vendor/github.com/spf13/cobra/go.sum
29:github.com/coreos/bbolt v1.3.2/go.mod h1:iRUV2dpdMOn7Bo10OQBFzIJO9kkE559Wcmn+qkEiiKk=
30:github.com/coreos/etcd v3.3.13+incompatible/go.mod h1:uF7uidLiAD3TWHmW31ZFd/JWoc32PjwdhPthX9715RE=
186:go.etcd.io/bbolt v1.3.2/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=

./vendor/github.com/spf13/cobra/projects_using_cobra.md
9:- [Etcd](https://etcd.io/)

./go.sum
54:github.com/coreos/bbolt v1.3.2 h1:wZwiHHUieZCquLkDL0B8UhzreNWsPHooDAG3q34zk0s=
55:github.com/coreos/bbolt v1.3.2/go.mod h1:iRUV2dpdMOn7Bo10OQBFzIJO9kkE559Wcmn+qkEiiKk=
56:github.com/coreos/etcd v3.3.13+incompatible h1:8F3hqu9fGYLBifCmRCJsicFqDx/D68Rt3q1JMazcgBQ=
57:github.com/coreos/etcd v3.3.13+incompatible/go.mod h1:uF7uidLiAD3TWHmW31ZFd/JWoc32PjwdhPthX9715RE=
376:go.etcd.io/bbolt v1.3.2 h1:Z/90sZLPOeCy2PwprqkFa25PdkusRzaj9P8zm/KNyvk=
377:go.etcd.io/bbolt v1.3.2/go.mod h1:IbVyRI1SCnLcuJnV2u8VeU0CEYM7e686BmAb1XKL+uU=
```

They are literally never mentioned anywhere besides the `go.sum` files, they aren't actually imported by anything, even our dependencies' dependencies... :confused:

File: go.sum (hunk 1 above)
Comment (MStoykov) [reply go.sum:8]: I think I forgot to mention this .. but it is :drum: https://github.com/spf13/viper , which imports most things IMO:
```
$ go mod why -m cloud.google.com/go
# cloud.google.com/go
github.com/spf13/viper/remote
github.com/bketelsen/crypt/config
github.com/bketelsen/crypt/backend/firestore
cloud.google.com/go/firestore
cloud.google.com/go
```

The good think is that go.sum is just the list of all modules and their versions that anything in the dependency list has sait it needs not stuff that we actually add to vendor or compile into the binary ... so this is mostly noise

File: go.sum (hunk 2 above)
Comment (MStoykov) [reply go.sum:54]: yes `https://github.com/spf13/viper` again which is only imported by `https://github.com/spf13/cobra` as it is used for it's own command line tools ... 

```
$ go mod why go.etcd.io/bbolt
# go.etcd.io/bbolt
github.com/spf13/viper/remote
github.com/bketelsen/crypt/config
github.com/bketelsen/crypt/backend/etcd
github.com/coreos/etcd/client
github.com/coreos/etcd/client.test
github.com/coreos/etcd/integration
github.com/coreos/etcd/etcdserver
github.com/coreos/etcd/mvcc/backend
github.com/coreos/bbolt
github.com/coreos/bbolt.test
go.etcd.io/bbolt
```

File: go.sum (hunk 1 above)
Comment (na--) [reply go.sum:8]: https://pkg.go.dev/github.com/spf13/viper#readme-remote-key-value-store-support :man_facepalming: 
Have I mentioned how much I ":heart:" Go modules recently?

File: go.sum (hunk 2 above)
Comment (na--) [reply go.sum:54]: Same reason as the previous comment apparently :face_with_head_bandage: https://pkg.go.dev/github.com/spf13/viper#readme-remote-key-value-store-support

Comment (na--) [reply go.sum:54]: but yeah, this is ridiculous - only `github.com/spf13/cobra/cobra/cmd` imports `github.com/spf13/viper`, which in turn imports all of this bloat, but we don't even import `github.com/spf13/cobra/cobra/cmd` in k6 :sob:

**Review by na-- — APPROVED** [review]
LGTM, though I hate the `go.sum` noise... :disappointed: 

I also think you'd also inadvertently updated `github.com/julienschmidt/httprouter` in this PR, which doesn't seem to be a dependency of the cobra or color libraries... So can you please update the PR description, or rebase the PR without its updates?

**Comment by na--** [comment]
Hmm just saw in the description of https://github.com/k6io/k6/pull/1947 that you maybe plan to completely drop `httprouter` in the future, so I guess we shouldn't update it here at all

**Comment by MStoykov** [comment]
It is updated here because:
1. we use it currently
2. cobra ... imports ... probably through viper and requires a higher version

I can 'replace' it, if you want, but the update is 16 lines in total so I think it will be just easier to leave it as is and if we manage to drop it - drop it at a later point

**Review by na-- — APPROVED** [review]
Fair enough
